openfisca_core/formulas.py

```diff
@@ -774,15 +770,14 @@ def new_filled_column(
     elif isinstance(source_file_path, str):
         source_file_path = source_file_path.decode('utf-8')
 
-    if start_date is UnboundLocalError:
-        start_date = None if reference_column is None else reference_column.start
-    elif start_date is not None:
-        assert isinstance(start_date, datetime.date)
-
-    if stop_date is UnboundLocalError:
-        stop_date = None if reference_column is None else reference_column.end
-    elif stop_date is not None:
-        assert isinstance(stop_date, datetime.date)
+    if end is UnboundLocalError:
+        end = None if reference_column is None else reference_column.end
+    elif end is not None:
+        assert isinstance(end, str), 'Type error on {}. String expected. Found: {}'.format(name + '.end', type(end))
+        try:
+            get_datetime_date(end)
```

Comment:
I've been told you reached an agreement orally, glad to have been a good plastic duck 😄

I'm obviously in favour of limiting performance impact, if there is no good reason to multiply this parsing then let's not do it — duh.

I would like to bring your attention to the fact that it is more than time to ship, though. Too much energy and time is going into something that is not a priority. If it is too hard to reach a consensus on the implementation we'll have to abandon that story and learn from it. This is becoming too much of a drain on delivering value.
I'm sure you can get this done quickly. Please don't hesitate to take shortcuts, the quality of this code is already much above average. Better done than perfect.